tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -1774,6 +1774,11 @@ def map(self, map_func, num_parallel_calls=None, deterministic=None):
     in general precludes the possibility of executing user-defined
     transformations in parallel (because of Python GIL).
 
+    The order of elements yielded by this transformation is
```

Comment:
Please clarify that this paragraph applies to cases where `num_parallel_calls > 1`. When `num_parallel_calls <= 1`, the order is deterministic as long as the `map_func` is deterministic (doesn't need to be a pure func and `deterministic=True` isn't needed).